packages/types/src/payment-types.ts

```diff
@@ -54,12 +32,60 @@ export interface IFeeReferenceBasedCreationParameters extends IReferenceBasedCre
 }
 
 /** Parameters to create a request with "any to erc20" payment network */
-export interface IAnyToErc20CreationParameters extends IFeeReferenceBasedCreationParameters {
-  network?: string;
+export interface IAnyToErc20CreationParameters extends ICreationParametersAnyToAny {
   acceptedTokens?: string[];
-  maxRateTimespan?: number;
 }
 
+/** Interface to create a payment network  */
+export interface IPaymentNetworkCreateParameters<T = any> {
+  id: PAYMENT_NETWORK_ID;
+  parameters: T;
+}
+
+export type PaymentNetworkCreateParameters =
+  | {
+      id:
+        | ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_PROXY_CONTRACT
+        | ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT
+        | ExtensionTypes.PAYMENT_NETWORK_ID.ETH_FEE_PROXY_CONTRACT
+        | ExtensionTypes.PAYMENT_NETWORK_ID.ETH_INPUT_DATA;
+      parameters: ExtensionTypes.PnFeeReferenceBased.ICreationParameters;
+    }
+  | {
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.ANY_DECLARATIVE;
+      parameters: ExtensionTypes.PnAnyDeclarative.ICreationParameters;
+    }
+  | {
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY;
+      parameters: ExtensionTypes.PnAnyToErc20.ICreationParameters;
+    }
+  | {
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.ANY_TO_ETH_PROXY;
+      parameters: ExtensionTypes.PnAnyToEth.ICreationParameters;
+    }
+  | {
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.ANY_TO_NATIVE_TOKEN;
+      parameters: ExtensionTypes.PnAnyToAnyConversion.ICreationParameters;
+    }
+  | {
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.NATIVE_TOKEN;
+      parameters: ExtensionTypes.PnReferenceBased.ICreationParameters;
+    }
+  | {
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.ERC777_STREAM;
+      parameters: ExtensionTypes.PnStreamReferenceBased.ICreationParameters;
+    }
+  | {
+      id:
+        | ExtensionTypes.PAYMENT_NETWORK_ID.BITCOIN_ADDRESS_BASED
+        | ExtensionTypes.PAYMENT_NETWORK_ID.TESTNET_BITCOIN_ADDRESS_BASED
+        | ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_ADDRESS_BASED;
+      parameters: ExtensionTypes.PnAddressBased.ICreationParameters;
+    };
+
+/** Payment network IDs with known Create Parameters */
+export type PAYMENT_NETWORK_ID = PaymentNetworkCreateParameters['id'];
```

Comment:
Isn't this the same as the new type from extension-types?
`export type ID = EnumToType<typeof ID>;`

Reply:
I really hesitated to keep it, you're right I think, I'll remove it. Thanks @alexandre-abrioux